File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/schema/content/ContentTypeUpdatedEvent.ts
module api.schema.content {

    export class ContentTypeUpdatedEvent extends api.event.Event {

        private name: ContentTypeName;

        private modifiedTime: Date;

        constructor(name: ContentTypeName, modifiedTime: Date) {
            super();
            this.name = name;
            this.modifiedTime = modifiedTime;
        }

        getContentTypeName(): ContentTypeName {
            return this.name;
        }

        getModifiedTime(): Date {
            return this.modifiedTime;
        }

        static on(handler: (event: ContentTypeUpdatedEvent) => void) {
            api.event.Event.bind(api.util.getFullName(this), handler);
        }

        static un(handler?: (event: ContentTypeUpdatedEvent) => void) {
            api.event.Event.unbind(api.util.getFullName(this), handler);
        }


        static fromJson(json: ContentTypeUpdatedEventJson): ContentTypeUpdatedEvent {
            return new ContentTypeUpdatedEvent(new ContentTypeName(json.name), json.modifiedTime);
        }
    }

    export interface ContentTypeUpdatedEventJson {

        name: string;

        modifiedTime: Date;
    }
}

File: modules/core-api/src/main/java/com/enonic/xp/core/schema/content/ContentTypeUpdatedEvent.java
package com.enonic.xp.core.schema.content;

import java.time.Instant;

import com.enonic.xp.core.event.Event;

public class ContentTypeUpdatedEvent
    implements Event
{
    private final ContentTypeName name;

    private final Instant modifiedTime;

    public ContentTypeUpdatedEvent( final ContentTypeName name, final Instant modifiedTime )
    {
        this.name = name;
        this.modifiedTime = modifiedTime;
    }

    public ContentTypeName getName()
    {
        return name;
    }

    public Instant getModifiedTime()
    {
        return modifiedTime;
    }
}
